app/configurator/components/chart-configurator.tsx

```diff
@@ -177,23 +177,52 @@ export const ChartConfigurator = ({
     });
   }, [variables, executeQuery]);
 
+  const client = useClient();
+
   useEffect(() => {
-    if (!metaData || !data || !data.dataCubeByIri) {
+    if (!metaData) {
       return;
     }
-    // Make sure that the filters are in line with the values
-    const chartConfig = ensureFilterValuesCorrect(state.chartConfig, {
-      dimensions: data.dataCubeByIri.dimensions,
-    });
 
-    dispatch({
-      type: "CHART_CONFIG_REPLACED",
-      value: {
-        chartConfig,
-        dataSetMetadata: metaData,
-      },
-    });
-  }, [data, dispatch, metaData, state.chartConfig]);
+    const run = async () => {
+      const {
+        data,
+        error,
+      }: { data?: PossibleFiltersQuery; error?: CombinedError } = await client
+        .query(PossibleFiltersDocument, {
+          iri: state.dataSet,
+          filters: state.chartConfig.filters,
+        })
+        .toPromise();
```

Comment:
I think that we use a `client` directly only in RDF queries and useXYZQuery elsewhere (`usePossibleFiltersQuery` hook in this case). Not sure why this pattern has changed here? Would the results still be cached?

Reply:
Mmh I think using useQuery here is not the clearer thing since what we want is to query the possible filters on filters change, and then possibly change the configuration. I tried just now to use usePossibleFiltersQuery + useEffect and it did not work well as the configuration was reset to what was there before as soon as I make a change (since the query is not yet changed, but configuration has changed, the effect is run, and the filters are reset according to the *previous* filters that have been chosen.

I think the imperative logic here is easier to follow.

Reply:
And for the caching, it works according to documentation

> By default, urql uses a concept called Document Caching. It will avoid sending the same requests to a GraphQL API repeatedly by caching the result of each query.

Reply:
👍